### DISCUSSION TITLE
No LSP for Python after fresh installation

### DISCUSSION THREAD

**Discussion by hopezh** [LazyVim/LazyVim#3956] [Q&A]
Using "K" (hover) to show definition of classes or methods is not working for `python` file in a fresh installation of `lazyvim`, although a `pyright` is attached to the current buffer, as shown below.

May I ask what could be causing this issue?  

Thanks.

1. `:LspInfo`
<img width="1428" alt="Screenshot 2024-07-08 at 10 28 10 PM" src="https://github.com/LazyVim/LazyVim/assets/23435810/b51a315f-5cf1-4250-bef1-3dc8d22b5642">

2. `:Mason`
<img width="1441" alt="Screenshot 2024-07-08 at 10 37 22 PM" src="https://github.com/LazyVim/LazyVim/assets/23435810/bf47dc5a-83ff-4d05-a5d3-97056efbb16f">

3. `:checkhealth` (lsp section)
```
vim.lsp: require("vim.lsp.health").check()

- LSP log level : WARN
- Log path: /Users/oatmini/.local/state/oatVim06/lsp.log
- Log size: 14 KB

vim.lsp: Active Clients ~
- pyright (id=1, root_dir=~, attached_to=[49])
- ruff_lsp (id=2, root_dir=~/.config/oatVim06/v:null, attached_to=[49])

vim.lsp: File watcher ~
- File watch backend: libuv-watch
```

**Comment by ddhanak** [comment]
same issue here, not sure what is wrong 🤔

**Comment by dpetka2001** [comment]
Maybe related? https://github.com/microsoft/pyright/issues/2135

**Reply by hopezh** [reply]
Sorry, I don't see how it's related... The issue remains after updating lazyvim...

**Comment by hopezh** [comment]
I'm still having this issue: unable to use "K (hover)" to show lsp info for python code in a freshly installed lazyvim, although "pyright" is attached to the buffer as shown by "LspInfo" command. 

Anyone has similar issue here?  

<img width="2904" alt="Screenshot 2024-07-19 at 10 57 24 PM" src="https://github.com/user-attachments/assets/c396ac36-1e99-4fb3-aea7-99b94d3578e5">

**Reply by dpetka2001** [reply]
Full screencast of default LazyVim installation and Python virtual environment with `pandas` installed based on your example screenshot. It seems to work correctly for me. I hope I didn't misunderstand what you meant. If I did misunderstood you please explain in more details what you meant.

[Screencast 2024-07-19 18:29:33.webm](https://github.com/user-attachments/assets/8a07e792-d591-4eaa-a5d8-62df8a410eee)

**Reply by hopezh** [reply]
Thanks, @dpetka2001. The video below shows the issue I have.

I did noticed that this only happens on my Mac Mini computer with Mac OS, not on my Windows laptop. I don't know why this is the case here.

1. video 1 shows the fresh installation of lazyvim:
https://1drv.ms/v/s!AjuL8pjKs_fCo_pqaCsHu8rK6dTeYg?e=yzOvRj

2. video 2 shows the activation of python support via lazyExtra: 
https://1drv.ms/v/s!AjuL8pjKs_fCo_ppoBMOG-G2aD26UQ?e=nvg769

**Reply by dpetka2001** [reply]
Try to create a Python virtual env and install pandas there and open Neovim from inside the virtual env to see how it behaves. Maybe you would also like to set `:h g:python3_host_prog` to see if it makes any difference?

**Reply by hopezh** [reply]
I've configured `vim.g.python3_host_prog` and I still don't have "K (hover" for python code.

<img width="1443" alt="Screenshot 2024-07-20 at 8 24 49 PM" src="https://github.com/user-attachments/assets/a65af583-ac43-4d2b-bbba-e59faba31352">

---

<img width="1443" alt="Screenshot 2024-07-20 at 8 25 49 PM" src="https://github.com/user-attachments/assets/b20eec81-89c8-4c5b-85cf-fd02651302cd">

---

Anyway, I tried the [`SciVim`](https://github.com/Dan7h3x/SciVim) neovim config by [Dan7h3x](https://github.com/Dan7h3x), and the `K (hover)` and all other LSP functions work well. 

<img width="1448" alt="Screenshot 2024-07-20 at 8 31 11 PM" src="https://github.com/user-attachments/assets/dfed1849-5386-4a2a-85f8-98b9e39180c6">


So, I think sth is wrong with my OS, and it's NOT a `lazyvim` issue. I have to continue digging the source of the problem...

**Reply by wulymammoth** [reply]
LSP hover used to work for me as well, but after an update in the last two weeks, I no longer have this as well, and I've attempted to debug to no avail. I'm also on MacOS

**Reply by hopezh** [reply]
Hi @wulymammoth, after updating my nvim config to the latest lazyvim, and that of many other plugins, the issue I reported is gone, strangely... I got the "K (hover" function back, and I still have no idea of the source of the original issue.  (I was about to give up and already started working a new config from scratch...)

1. Here's the image of my updated config: [oatvim06](https://github.com/hopezh/oatVim06):

<img width="1444" alt="Screenshot 2024-08-05 at 11 19 37 PM" src="https://github.com/user-attachments/assets/7830aab5-2442-486b-a24c-a121f414ff61">

2. Here's the image of a fresh lazyvim installation:

<img width="1444" alt="Screenshot 2024-08-05 at 11 21 22 PM" src="https://github.com/user-attachments/assets/427ab0d7-3f5d-41d0-b777-bba776041f2d">
